src/jdk.attach/macosx/native/libattach/VirtualMachineImpl.c

```diff
@@ -123,38 +124,39 @@ JNIEXPORT jboolean JNICALL Java_sun_tools_attach_VirtualMachineImpl_checkCatches
     size_t            kipsz = sizeof(struct kinfo_proc);
 
    /*
-    * early in the lifetime of a JVM it has not yet initialized its signal handlers, in particular the QUIT
+    * Early in the lifetime of a JVM it has not yet initialized its signal handlers, in particular the QUIT
     * handler, note that the default behavior of QUIT is to terminate the receiving process, if unhandled.
     *
-    * since we use QUIT to initiate an attach operation, if we signal a JVM during this period early in its
+    * Since we use QUIT to initiate an attach operation, if we signal a JVM during this period early in its
     * lifetime before it has initialized its QUIT handler, such a signal delivery will terminate the JVM we
     * are attempting to attach to!
     *
-    * the following code guards the QUIT delivery by testing the current signal masks
+    * The following code guards the QUIT delivery by testing the current signal masks. It is okay to send QUIT
+    * if the signal is caught but not ignored, as that implies a handler has been installed.
     */
 
     if (sysctl(mib, sizeof(mib) / sizeof(int), &kiproc, &kipsz, NULL, 0) == 0) {
-        const int ignored = (kiproc.kp_proc.p_sigignore & sigmask(SIGQUIT)) != 0;
-        const int caught  = (kiproc.kp_proc.p_sigcatch & sigmask(SIGQUIT))  != 0;
+        const bool ignored = (kiproc.kp_proc.p_sigignore & sigmask(SIGQUIT)) != 0;
+        const bool caught  = (kiproc.kp_proc.p_sigcatch & sigmask(SIGQUIT))  != 0;
 
         // *only* send QUIT if the target is ready to catch and handle the signal to avoid default "death" if not
 
         // note: obviously the masks could change between testing and signalling however this is not the
         // observed behavior of the current JVM implementation.
 
         if (caught && !ignored) {
-    	    if (kill((pid_t)pid, SIGQUIT)) {
+            if (kill((pid_t)pid, SIGQUIT)) {
```

Comment:
Style nit: no implicit booleans
```suggestion
            if (kill((pid_t)pid, SIGQUIT) != 0) {
```

Reply:
note that I did not modify this line of code, it is as it was prior to my changes!

closing the stable door after the horse has bolted!m

Reply:
Noted. But good to clean up whilst making significant code changes. Thanks